### PR TITLE
Schema support for plot

### LINKED ISSUE #854: Can't denote DuckDB table with custom schema using %sqlplot
I've not been able to execute %sqlplot referencing  a DuckDB table or view with a custom schema.

Ideally tables and views within a custom schemas could be referenced either automatically with schema_name.table_name syntax or with the addition of a --schema argument. (Or both)

Table with custom schema:
image

### PR DESCRIPTION
**Pull request by bbeat2782**
## Describe your changes

Add `schema/-s` for `%sqlplot` and ensure `--table schema.table` works

## Issue number

Closes #854 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [x] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [x] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--855.org.readthedocs.build/en/855/

### CODE REVIEW

File: src/sql/util.py [hunk 1]
```diff
@@ -404,3 +405,19 @@ def if_substring_exists(string, substrings):
     """Function to check if any of substring in
     substrings exist in string"""
     return any(msg in string for msg in substrings)
+
+
+def enclose_table_with_double_quotations(table):
```
Comment (edublancas) [inline src/sql/util.py:410]: not all databases use double quotes to identify table names, we already have some utility to detect this: https://github.com/search?q=repo%3Aploomber%2Fjupysql%20is_use_backtick_template&type=code

Comment (edublancas) [reply src/sql/util.py:410]: please use the is_use_backtick_template function to determine which character to use

File: src/tests/integration/test_generic_db_operations.py [hunk 2]
```diff
@@ -477,6 +477,68 @@ def test_sqlplot_pie(ip_with_dynamic_db, request, test_table_name_dict):
...
+            marks=pytest.mark.xfail(reason="sqlplot does not support SQL server"),
+        ),
+        pytest.param(
+            "ip_with_clickhouse",
+            marks=pytest.mark.xfail(
+                reason="Plotting from snippet not working in clickhouse"
+            ),
+        ),
+    ],
+)
```
Comment (edublancas) [inline src/tests/integration/test_generic_db_operations.py:511]: please open a new issue pointing to these tests so we tackle them later

**Review by edublancas — CHANGES_REQUESTED** [review]

**Review by neelasha23 — DISMISSED** [review]
Tested locally and it works fine!

File: src/sql/util.py (hunk 1 above)
Comment (bbeat2782) [reply src/sql/util.py]: [fixed](https://github.com/ploomber/jupysql/pull/855/files#diff-653c15e6161c97f356be04e072838ed15014af275c06019a7692ab727fa44b97R423-R425)

**Comment by edublancas** [comment]
is this ready for review @bbeat2782 ?

**Comment by bbeat2782** [comment]
@edublancas I was going to request a re-review after doing some testing on the failed test cases so that I can add more information to the issue, but besides that, it is ready for review

File: src/tests/integration/test_generic_db_operations.py (hunk 2 above)
Comment (bbeat2782) [reply src/tests/integration/test_generic_db_operations.py:511]: issue about the failing test cases : [here](https://github.com/ploomber/jupysql/issues/866)

**Review by edublancas — APPROVED** [review]

**Pull request merged**